tests/integration/zipkin_integration_test.py

```diff
@@ -90,19 +172,24 @@ def test_starting_zipkin_trace_with_128bit_trace_id(
         binary_annotations={'some_key': 'some_value'},
         add_logging_annotation=True,
         use_128bit_trace_id=True,
-        firehose_handler=mock_firehose_handler
+        firehose_handler=mock_firehose_handler,
+        encoding=encoding,
     ):
         pass
 
     def check_span(span):
         assert span.trace_id is not None
-        assert span.trace_id_high is not None
+        if encoding == py_zipkin.Encoding.THRIFT:
+            assert span.trace_id_high is not None
+        elif encoding == py_zipkin.Encoding.JSON:
+            assert len(span.trace_id) == 32
 
-    check_span(_decode_binary_thrift_obj(mock_logs[0]))
-    check_span(_decode_binary_thrift_obj(mock_firehose_logs[0]))
+    check_span(decode(mock_logs[0], encoding)[0])
+    check_span(decode(mock_firehose_logs[0], encoding)[0])
 
 
-def test_span_inside_trace():
+@pytest.mark.parametrize('encoding', SUPPORTED_ENCODINGS)
```

Comment:
I'm not sure if you want to go down this route (i'm thinking that it might actually make things more complicated), but it looks like pytest supports test-suite level parametrization: https://docs.pytest.org/en/latest/example/parametrize.html#a-quick-port-of-testscenarios